src/status_im/contexts/communities/events.cljs

```diff
@@ -163,12 +163,15 @@
 (defn initialize-permission-addresses
   [{:keys [db]} [community-id]]
   (when community-id
-    (let [accounts        (get-in db [:wallet :accounts])
-          sorted-accounts (sort-by :position (vals accounts))
-          addresses       (set (map :address sorted-accounts))]
+    (let [accounts         (get-in db [:wallet :accounts])
+          sorted-accounts  (sort-by :position (vals accounts))
+          watched-accounts (remove #(:watch-only? %) sorted-accounts)
```

Comment:
I have the same logic in three different events. Can you provide guidance on how to make it reusable? (1)

Reply:
a function that takes the `db` or the accounts and returns the `addresses` correctly ordererd?